src/ucp/rma/flush.c

```diff
@@ -310,26 +315,82 @@ static ucs_status_t ucp_worker_flush_check(ucp_worker_h worker)
 
         status = uct_iface_flush(worker->ifaces[rsc_index].iface, 0, NULL);
         if (status != UCS_OK) {
+            if (UCS_STATUS_IS_ERR(status)) {
+                ucs_error("iface[%d] "UCT_TL_RESOURCE_DESC_FMT" flush failed: %s",
+                          rsc_index,
+                          UCT_TL_RESOURCE_DESC_ARG(&worker->context->tl_rscs[rsc_index].tl_rsc),
+                          ucs_status_string(status));
+            }
             return status;
         }
     }
 
     return UCS_OK;
 }
 
-static unsigned ucp_worker_flush_progress(void *arg)
+static void ucp_worker_flush_complete_one(ucp_request_t *req, ucs_status_t status,
+                                          int force_progress_unreg)
 {
-    ucp_request_t *req  = arg;
     ucp_worker_h worker = req->flush_worker.worker;
+
+    if (force_progress_unreg) {
```

Comment:
maybe add it to the `else` branch of the `if` below?
Or move it to the end of func and return inside the `if` below
Otherwise `uct_worker_progress_unregister_safe` can be called twice, which should not result in an error, but can be avoded